### PR TITLE
Expose the pprof server in the operator

### LINKED ISSUE #1997: Expose the pprof server in the operator
The [`controller-runtime`](https://github.com/kubernetes-sigs/controller-runtime/pull/1943) dependency added a `pprof` server flag that can be enabled to get information from the operator.

This issue is to add support to the operator to expose that information. The feature should be enabled just for debugging purposes.

### PR DESCRIPTION
**Pull request by iblancasa**
Resolves #1997

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -107,6 +108,7 @@ func main() {
 
 	pflag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
 	pflag.StringVar(&probeAddr, "health-probe-addr", ":8081", "The address the probe endpoint binds to.")
+	pflag.StringVar(&pprofAddr, "pprof-addr", "", "The address to expose the pprof server.")
```
Comment (jaronoff97) [inline main.go:111]: should we set a default here?

Comment (iblancasa) [reply main.go:111]: The default is an empty string. When the empty string is provided to the manager, the server is not started (the default).

Comment (jaronoff97) [reply main.go:111]: can we add that to the docstring here? something like "default is empty string which disables the pprof server"

File: .chloggen/1997-expose-pprof-server.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. operator, target allocator, github action)
+component: operator
+
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Add support to expose the pprof server in the operator
```
Comment (frzifus) [inline .chloggen/1997-expose-pprof-server.yaml:8]:
```suggestion
note: Add flag to enable support for the pprof server in the operator.
```

File: main.go (hunk 1 above)
Comment (iblancasa) [reply main.go:111]: Sure!

**Pull request merged**